### PR TITLE
feat: add debug logging

### PR DESCRIPTION
**Pull request by bajtos**

### CODE REVIEW

File: lib/evaluate.js [hunk 1]
```diff
@@ -82,6 +86,8 @@ export const evaluate = async ({
 
 export const runFraudDetection = async (roundIndex, measurements, { fetchRoundDetails, recordTelemetry }) => {
   const roundDetails = await fetchRoundDetails(roundIndex, recordTelemetry)
+  debug(`ROUND DETAILS for round=${roundIndex}`, roundDetails)
```
Comment (juliangruber) [inline lib/evaluate.js:89]:
```suggestion
  debug(`ROUND DETAILS for round=%s`, roundIndex, roundDetails)
```

`debug` wants to use printf formatting, let's be consistent

File: lib/evaluate.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
+import debugFactory from 'debug'
+
+const debug = debugFactory('spark:evaluate')
```
Comment (juliangruber) [inline lib/evaluate.js:3]:
```suggestion
import createDebug from 'debug'

const debug = createDebug('spark:evaluate')
```

Not important, but I would prefer this naming

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: lib/evaluate.js (hunk 2 above)
Comment (bajtos) [reply lib/evaluate.js:3]: I like that 👍🏻

**Comment by bajtos** [comment]
@juliangruber I addressed both your comments, LGTY now?

File: lib/evaluate.js [hunk 3]
```diff
@@ -82,6 +86,9 @@ export const evaluate = async ({
 
 export const runFraudDetection = async (roundIndex, measurements, { fetchRoundDetails, recordTelemetry }) => {
   const roundDetails = await fetchRoundDetails(roundIndex, recordTelemetry)
+  // IMPORTANT: The format string MUST NOT have any placeholder for the roundDetails object
```
Comment (juliangruber) [inline lib/evaluate.js:89]:
```suggestion
  // Omit the roundDetails object from the format string to get nicer formatting
```

Less screamy plus explains why

File: lib/evaluate.js [hunk 4]
```diff
@@ -95,4 +102,17 @@ export const runFraudDetection = async (roundIndex, measurements, { fetchRoundDe
 
     m.fraudAssessment = 'OK'
   }
+
+  if (debug.enabled) {
+    for (const m of measurements) {
+      // Print round & participant address & CID together to simplify lookup when debugging
+      // IMPORTANT: The format string MUST NOT have any placeholder for the `m` object
```
Comment (juliangruber) [inline lib/evaluate.js:109]:
```suggestion
      // Omit the `m` object from the format string to get nicer formatting
```

**Review by juliangruber — APPROVED** [review]
Approved with tiny comment suggestions

**Pull request merged**